api/main_endpoints/routes/User.js

```diff
@@ -199,6 +207,73 @@ router.post('/getPagesPrintedCount', (req, res) => {
   });
 });
 
+// Verifies the users session if they have an active jwtToken.
+// Used on the inital load of root '/'
+// Returns the name and accesslevel of the user w/ the given access token
+router.post('/verify', function(req, res) {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+
+  const decoded = checkIfTokenValid(req);
+  if (decoded) {
+    res.status(OK).send(decoded);
+  } else {
+    res.sendStatus(UNAUTHORIZED);
+  }
+});
+
+router.get('/callback', async function(req, res) {
+  const code = req.query.code;
+  const email = req.query.state;
```

Comment:
for these lines you can just send over params like `code` and `email` instead of `code` and `state`, then u can do

```js
const { code, email } = req.query;
```

read about [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

Reply:
tried and doesn't work that way, email and state are differently named